apiserver/facades/client/application/deployrepository_test.go

```diff
@@ -685,6 +693,44 @@ func (s *deployRepositorySuite) TestDeployFromRepositoryAPI(c *gc.C) {
 	})
 }
 
+// The reason for this matcher is that the AddApplicationArgs.Charm is
+// obtained by casting application.Charm into a state.Charm, but we
+// can't do that cast with a MockCharm
+type addApplicationArgsMatcher struct {
+	c            *gc.C
+	expectedArgs state.AddApplicationArgs
+}
+
+func (m addApplicationArgsMatcher) String() string {
+	return "match AddApplicationArgs"
+}
+
+func (m addApplicationArgsMatcher) Matches(x interface{}) bool {
+
+	oA, ok := x.(state.AddApplicationArgs)
+	if !ok {
+		return false
+	}
+
+	eA := m.expectedArgs
+	// Check everything but the Charm
+	check := oA.Name == eA.Name &&
+		oA.ApplicationConfig == eA.ApplicationConfig &&
+		oA.NumUnits == eA.NumUnits &&
+		oA.Constraints == eA.Constraints
+
+	checkStorage, _ := jc.DeepEqual(oA.Storage, eA.Storage)
+	checkDevices, _ := jc.DeepEqual(oA.Devices, eA.Devices)
+	checkAttachStorage, _ := jc.DeepEqual(eA.AttachStorage, eA.AttachStorage)
+	checkExportBindings, _ := jc.DeepEqual(oA.EndpointBindings, eA.EndpointBindings)
+	checkCharmConfig, _ := jc.DeepEqual(oA.CharmConfig, eA.CharmConfig)
+	checkPlacement, _ := jc.DeepEqual(oA.Placement, eA.Placement)
+	checkResources, _ := jc.DeepEqual(oA.Resources, eA.Resources)
+
+	return check && checkStorage && checkDevices && checkAttachStorage &&
+		checkExportBindings && checkCharmConfig && checkPlacement && checkResources
```

Comment:
While functionally correct, it'd would be impossible to debug a test failure without adding code. Please add a way to find what didn't pass the check easily.